tests/iio_attr.c

```diff
@@ -154,6 +176,16 @@ static void dump_device_attributes(const struct iio_device *dev,
 		if (!quiet)
 			printf("dev '%s', attr '%s', value :",
 					iio_device_get_name(dev), attr);
+		if(fd) {
+			fprintf(fd, "\t/* Read IIO device Attribute, and put result in string */\n"
```

Comment:
generally: i am thinking that it may make sense to have a generic helper (something like `iio_cgen_line()` or whatever name) that is `fprintf()` wrapper;

there's a few things that this could address:
1. the `\n` char at the end of each line
2. indentation levels, which would print multiple `\t` at the beginning of each line;

this helper can be a inline function or a precompiler macro; i guess it does not matter much what it is;
but removing some elements that are prone to minor errors [or styling] when generating code, would be an idea;

examples:
```
iio_cgen_line(fd, 1, "/* Read IIO device Attribute, and put result in string */")   // one indentation level
iio_cgen_line(fd, 2, "/* Read IIO device Attribute, and put result in string */")   // 2 indentation levels
// or
iio_cgen_line1(fd, "/* Read IIO device Attribute, and put result in string */")   // one indentation level
iio_cgen_line2(fd, "/* Read IIO device Attribute, and put result in string */")   // 2 indentation levels
```

Reply:
personally - one line macro functions that just replace printf don't make it easier to read/understand the code - not sure this is a good idea or not.

There isn't any indentations but none/one. The code that it generates is all flat. There are no loops/if structures (and should never be).

Reply:
ack;
if stuff like looks/if structures become present, we can add these type of helpers then;
`iio_cgen_line(fd, indent_level_N,)`  type of funcs seemed useful when I was writing code that outputs pretty-formatted JSON [for whatever userland app I was writing], and I recalled this stuff when I was thinking about formatting this C code gen stuff;